modules/http-statistics-monitoring/src/main/java/org/wso2/appserver/monitoring/HttpStatValve.java

```diff
@@ -0,0 +1,205 @@
+    private String password = DefaultConfigurationConstants.PASSWORD;
+    private String configFileFolder = DefaultConfigurationConstants.CONFIG_FILE_FOLDER;
+    private String url = DefaultConfigurationConstants.URL;
+    private String streamId = DefaultConfigurationConstants.STREAM_ID;
+    private DataPublisher dataPublisher = null;
+    private String appServerHome;
+
+    @Override
+    protected void initInternal() {
+        LOG.debug("The HttpStatValve initialized.");
+        File userDir = new File(System.getProperty("catalina.base"));
+        appServerHome = userDir.getAbsolutePath();
+    }
+
+    @Override
+    public void invoke(Request request, Response response)  {
+
+        if (dataPublisher == null) {
+            try {
+                dataPublisher = getDataPublisher();
+            } catch (ConfigurationException e) {
+                LOG.error("Failed at setting configurations: " + e);
+            }
+        }
+
+        Long startTime = System.currentTimeMillis();
+
+        try {
+            getNext().invoke(request, response);
+        } catch (IOException e) {
+            LOG.error("IO error occurred:" + e);
+        } catch (ServletException e) {
+            LOG.error("Servlet error occurred" + e);
+        }
+        long responseTime = System.currentTimeMillis() - startTime;
+
+        Event event = null;
+        try {
+            event = EventBuilder.buildEvent(getStreamId(), request, response, startTime, responseTime);
+        } catch (EventBuilderException e) {
+            LOG.error("Creating the Event failed: " + e);
+        }
+        dataPublisher.publish(event);
+
+    }
+
+    //get file path to the file containing Data Agent configuration and properties
+    private String getDataAgentConfigPath() {
+        Path path = Paths.get(appServerHome, getConfigFileFolder(), "data-agent-conf.xml");
+        return path.toString();
+    }
+
+    //instantiate a data publisher to be used to publish data to DAS
+    private DataPublisher getDataPublisher() throws ConfigurationException {
+
+        AgentHolder.setConfigPath(getDataAgentConfigPath());
+        String url = getUrl();
+        DataPublisher dataPublisher = null;
+
+        try {
+            dataPublisher = new DataPublisher(url, getUsername(), getPassword());
+        } catch (DataEndpointAgentConfigurationException e) {
+            LOG.error("Data Endpoint Agent configuration failed: " + e);
+        } catch (DataEndpointException e) {
+            LOG.error("Communication with Data Endpoint failed: " + e);
+        } catch (DataEndpointConfigurationException e) {
+            LOG.error("Parsing Data Endpoint configurations failed: " + e);
+        } catch (DataEndpointAuthenticationException e) {
+            LOG.error("Connection to Data Endpoint failed during authentication: " + e);
+        } catch (TransportException e) {
+            LOG.error("Connection failed: " + e);
+        }
```

Comment:
This only logs the exception and continue. if there is an issue while initializing the data publisher, it should throw the exception.